pianoPlayer/main.js

```diff
@@ -0,0 +1,108 @@
+    event.target.style.top = '5px';
+    event.target.style.left = '5px';
+}
+function keyReturn(event) {
+    event.target.style.backgroundColor = '';
+    event.target.style.position = '';
+    event.target.style.top = '';
+    event.target.style.left = '';
+}
+function playNote(note) {
+    note.addEventListener('mousedown', keyPlay);
+    note.addEventListener('mouseup', keyReturn);
+}
+
+// loop that assigns the eventListeners to each key DOM element
+notes.forEach(playNote);
+
+// These variables store the buttons that progress the user through the lyrics
+let nextOne = document.getElementById('first-next-line');
+let nextTwo = document.getElementById('second-next-line');
+let nextThree = document.getElementById('third-next-line');
+let startOver = document.getElementById('fourth-next-line');
+
+// This variable stores the '-END' lyric element
+let lastLyric = document.getElementById('column-optional');
+
+// These statements are "hiding" all the progress buttons, but the first one
+nextTwo.hidden = true;
+nextThree.hidden = true;
+startOver.hidden = true;
+
+// anonymous event handler property and function for the first progress button
+nextOne.onclick = function () {
+    nextTwo.hidden = false;
+    nextOne.hidden = true;
+    document.getElementById('letter-note-five').innerHTML = 'D';
+    document.getElementById('letter-note-six').innerHTML = 'C';
+}
+
+// anonymous event handler property and function for the second progress button
+nextTwo.onclick = function () {
+    nextThree.hidden = false;
+    nextTwo.hidden = true;
+    document.getElementById('word-five').innerHTML = 'DEAR';
+    document.getElementById('word-six').innerHTML = 'FRI-';
+    lastLyric.style.display = 'inline-block';
+    document.getElementById('letter-note-three').innerHTML = 'G';
+    document.getElementById('letter-note-four').innerHTML = 'E';
+    document.getElementById('letter-note-five').innerHTML = 'C';
+    document.getElementById('letter-note-six').innerHTML = 'B';
+}
+
+// anonymous event handler property and function for the third progress button
+nextThree.onclick = function () {
+    startOver.hidden = false;
+    nextThree.hidden = true;
+    document.getElementById('word-one').innerHTML = 'HAP-';
+    document.getElementById('word-two').innerHTML = 'PY';
+    document.getElementById('word-three').innerHTML = 'BIRTH';
+    document.getElementById('word-four').innerHTML = 'DAY';
+    document.getElementById('word-five').innerHTML = 'TO';
+    document.getElementById('word-six').innerHTML = 'YOU!';
+    document.getElementById('letter-note-one').innerHTML = 'F';
+    document.getElementById('letter-note-two').innerHTML = 'F';
+    document.getElementById('letter-note-three').innerHTML = 'E';
+    document.getElementById('letter-note-four').innerHTML = 'C';
+    document.getElementById('letter-note-five').innerHTML = 'D';
+    document.getElementById('letter-note-six').innerHTML = 'C';
+    lastLyric.style.display = 'none';
+}
```

Comment:
nit: I would have all the document changes first (like setting the names) and then show the element after.

It's not a big deal, but it's a good habit to be in. It's sort of like how the curtain at a play doesn't go up until the stage and actors are all in place ready for the show to _start_.

Reply:
great analogy!